tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py

```diff
@@ -866,11 +866,11 @@ def __call__(self, inputs, state, scope=None):
           # Default initializer for embeddings should have variance=1.
           sqrt3 = math.sqrt(3)  # Uniform(-sqrt(3), sqrt(3)) has variance=1.
           initializer = init_ops.random_uniform_initializer(-sqrt3, sqrt3)
-
-        if type(state) is tuple:
-          data_type = state[0].dtype
-        else:
-          data_type = state.dtype
+          
+        s = state
```

Comment:
because of how wrappers tend to add their states at the end it might also be slightly faster to use s = s[-1] than s = s[0] but i don't think the state object is ever going to be deep enough for this to matter